# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1724013150 -7200
# Parent  240a252081dffcd31b902e8190d373ffde883afb
Misc: Change user agent string, take 2: Add Silicon/Intel for macOS.

diff --git a/mailnews/compose/src/MimeMessage.sys.mjs b/mailnews/compose/src/MimeMessage.sys.mjs
--- a/mailnews/compose/src/MimeMessage.sys.mjs
+++ b/mailnews/compose/src/MimeMessage.sys.mjs
@@ -190,8 +190,14 @@ export class MimeMessage {
             os = "Windows";
             break;
           case "macosx":
             os = "macOS";
+            const build = Services.sysinfo.getProperty("build").toLowerCase();
+            if (build.includes("_arm64_")) {
+              os += "/Silicon";
+            } else if (build.includes("_x86_")) {
+              os += "/Intel";
+            }
             break;
           case "linux":
             os = "Linux";
             break;
